internal/pkg/api/handleCheckin.go

```diff
@@ -855,6 +855,27 @@ func processPolicy(ctx context.Context, zlog zerolog.Logger, bulker bulk.Bulk, a
 	return &resp, nil
 }
 
+func getAgentAndVerifyAPIKeyID(ctx context.Context, bulker bulk.Bulk, agentID string, apiKeyID string) (*model.Agent, error) {
+	span, ctx := apm.StartSpan(ctx, "getAgentAndVerifyAPIKeyID", "read")
+	defer span.End()
+	agent, err := dl.GetAgent(ctx, bulker, agentID)
+	if err != nil {
+		if errors.Is(err, dl.ErrNotFound) {
+			err = ErrAgentNotFound
+		} else if errors.Is(err, dl.ErrNotFound) {
```

Comment:
Did you mean do have the same error twice?

Reply:
No that's a typo :)